src/components/PrintModal/WatermarkModal/WatermarkModal.js

```diff
@@ -0,0 +1,429 @@
+  TOP_RIGHT: 'Top Right',
+  TOP_CENTER: 'Top Center',
+  BOT_LEFT: 'Bottom Left',
+  BOT_RIGHT: 'Bottom Right',
+  BOT_CENTER: 'Bottom Center',
+};
+
+const FORM_FIELD_KEYS = {
+  location: 'location',
+  fontSize: 'fontSize',
+  text: 'text',
+  color: 'color',
+  opacity: 'opacity',
+};
+
+const DEFAULT_VALS = {
+  [FORM_FIELD_KEYS.location]: WATERMARK_LOCATIONS.CENTER,
+  [FORM_FIELD_KEYS.fontSize]: FONT_SIZES[FONT_SIZES.length / 2],
+  [FORM_FIELD_KEYS.text]: '',
+  [FORM_FIELD_KEYS.color]: new window.Annotations.Color(241, 160, 153),
+  [FORM_FIELD_KEYS.opacity]: 100,
+};
+
+class WatermarkModal extends React.PureComponent {
+  handleWatermarkRenderFxn;
+
+  static propTypes = {
+    isVisible: PropTypes.bool,
+    pageIndexToView: PropTypes.number,
+    modalClosed: PropTypes.func,
+    formSubmitted: PropTypes.func,
+    t: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const locationSettings = this.initializeLocationSettings();
+    this.state = {
+      isColorPaletteVisible: false,
+      locationSettings,
+      previousLocationSettings: locationSettings,
+    };
+    this.canvasContainerRef = React.createRef();
+
+    this.handleWatermarkRenderFxn = () => {
+      if (this.props.isVisible) {
+        this.setState({
+          locationSettings: this.state.previousLocationSettings,
+        }, () => {
+          this.addWatermarks();
+        });
+      } else {
+        this.removeAllWatermarks();
+      }
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.isVisible !== undefined) {
+      core.addEventListener('documentLoaded', this.handleWatermarkRenderFxn);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.isVisible !== prevProps.isVisible) {
+      this.setState({
+        isColorPaletteVisible: false,
+      }, () => this.handleWatermarkRenderFxn());
+    }
+  }
+
+  addWatermarks() {
+    const watermarkOptions = this.createWatermarks();
+
+    core.setWatermark(watermarkOptions);
+
+    const pageHeight = core.getPageHeight(this.props.pageIndexToView);
+    const pageWidth = core.getPageWidth(this.props.pageIndexToView);
+
+    const desiredZoomForWidth = DESIRED_WIDTH / pageWidth;
+    const desiredZoomForHeight = DESIRED_HEIGHT / pageHeight;
+
+    const desiredZoom = Math.min(desiredZoomForHeight, desiredZoomForWidth);
+
+    core.getDocument().loadCanvasAsync({
+      pageIndex: this.props.pageIndexToView,
+      zoom: desiredZoom,
+      drawComplete: canvas => {
+        const nodes = this.canvasContainerRef.current.childNodes;
+        if (nodes && nodes.length > 0) {
+          this.canvasContainerRef.current.removeChild(nodes[0]);
+        }
+        canvas.style.border = '1px solid black';
+        canvas.style.height = this.canvasContainerRef.current.style.height;
+        this.canvasContainerRef.current.appendChild(canvas);
+      },
+    });
+
+    // Note: do not update and refresh the doc else it may affect other docs as well
```

Comment:
What does this comment mean?

Reply:
Maybe its a poor comment on my end.

What i'm trying to say is that, developers should not add the following 2 lines:
window.docViewer.refreshAll();
window.docViewer.updateView();

Or else, the watermark can be seen in the other document:
![image](https://user-images.githubusercontent.com/54084743/64723348-3b2c2b00-d485-11e9-8f4a-6089a1f101d1.png)


I guess I should reword the comment for clarity?